src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -196,20 +196,21 @@ public UIThreadOnly(InteractiveWindow window, IInteractiveWindowEditorFactorySer
                 _outputTrackingCaretPosition = -1;
             }
 
-            public Task<ExecutionResult> ResetAsync(bool initialize)
+      
+            public Task<ExecutionResult> ResetAsync(bool initialize, bool isFromSubmit)
             {
                 Debug.Assert(State != State.Resetting, "The button should have been disabled.");
 
                 if (_window._stdInputStart != null)
                 {
-                     CancelStandardInput();
+                    CancelStandardInput();
                 }
 
                 _window._buffer.Flush();
 
                 // replace the task being interrupted by a "reset" task:
                 State = State.Resetting;
-                _currentTask = _window._evaluator.ResetAsync(initialize);
+                _currentTask = ((IInteractiveEvaluator2)(_window._evaluator)).ResetAsync(initialize, isFromSubmit);
```

Comment:
```_evaluator``` seems to be passed in via the public API ```IInteractiveWindowFactoryService.CreateWindow```.  Is it safe to assume that it's an ```IInteractiveEvaluator2```?  If so, then we should probably just change the type of the field.  Otherwise, we should have some fallback logic (i.e. pass ```isFromSubmit: true```).